readium/shared/src/main/java/org/readium/r2/shared/util/resource/ContentResource.kt

```diff
@@ -28,12 +28,24 @@ public class ContentResourceFactory(
     private val contentResolver: ContentResolver
 ) : ResourceFactory {
 
-    override suspend fun create(url: AbsoluteUrl): Try<Resource, ResourceFactory.Error> {
+    override suspend fun create(url: AbsoluteUrl): Resource? {
+        if (!url.isContent) {
+            return null
+        }
+
+        // FIXME: should check if uri points t o a file
+        return ContentResource(url.toUri(), contentResolver)
+    }
+
+    override suspend fun create(
+        url: AbsoluteUrl,
+        mediaType: MediaType
+    ): Try<Resource, ResourceFactory.Error> {
         if (!url.isContent) {
             return Try.failure(ResourceFactory.Error.SchemeNotSupported(url.scheme))
         }
 
-        val resource = ContentResource(url.toUri(), contentResolver)
+        val resource = ContentResource(url.toUri(), contentResolver, mediaType)
```

Comment:
`mediaType` is not used in most implementations of `ResourceFactory`. I think the other implementations should do like `ContentResourceFactory` and pass it to the implementations of `Resource`.

Reply:
`FileResourceFactory` does it as well... The only one which doesn't do it is the `HttpResourceFactory`, because I did't want to override the media type provided by the HTTP server.